src/storage/core.rs

```diff
@@ -1220,3 +1228,71 @@ where
         active + closed
     }
 }
+
+/// Key for demonstration purposes
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct UnitKey(Vec<u8>);
+
+impl AsRef<UnitKey> for UnitKey {
+    fn as_ref(&self) -> &UnitKey {
+        self
+    }
+}
+
+impl Default for UnitKey {
+    fn default() -> Self {
+        Self([0].to_vec())
+    }
+}
+
+impl AsRef<[u8]> for UnitKey {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl From<Vec<u8>> for UnitKey {
+    fn from(v: Vec<u8>) -> Self {
+        Self(v)
+    }
+}
+
+impl PartialOrd for UnitKey {
+    fn partial_cmp(&self, _: &Self) -> Option<CmpOrdering> {
+        Some(CmpOrdering::Equal)
```

Comment:
Just in case: can you implement trivial, but working key comaprison?

Reply:
Moreover, I think it is good idea to move `Key`, `RefKey` traits and this `UnitKey` into separate file and make `UnitKey` more universal (add generic parameter, that specifies key size)